pkg/gatherers/conditional/conditional_gatherer.go

```diff
@@ -88,18 +41,29 @@ type Gatherer struct {
 	metricsGatherKubeConfig *rest.Config
 	imageKubeConfig         *rest.Config
 	firingAlerts            map[string]bool // golang doesn't have sets :(
+	gatheringRulesEndpoint  string
+	gatheringRules          []GatheringRule
+	httpClient              *http.Client
 }
 
 // New creates a new instance of conditional gatherer with the appropriate configs
-func New(gatherProtoKubeConfig, metricsGatherKubeConfig *rest.Config) *Gatherer {
-	imageKubeConfig := rest.CopyConfig(gatherProtoKubeConfig)
-	imageKubeConfig.QPS = common.ImageConfigQPS
-	imageKubeConfig.Burst = common.ImageConfigBurst
+func New(gatherProtoKubeConfig, metricsGatherKubeConfig *rest.Config, gatheringRulesEndpoint string) *Gatherer {
+	var imageKubeConfig *rest.Config
+	if gatherProtoKubeConfig != nil {
+		imageKubeConfig = rest.CopyConfig(gatherProtoKubeConfig)
+		imageKubeConfig.QPS = common.ImageConfigQPS
+		imageKubeConfig.Burst = common.ImageConfigBurst
+	}
 
 	return &Gatherer{
 		gatherProtoKubeConfig:   gatherProtoKubeConfig,
 		metricsGatherKubeConfig: metricsGatherKubeConfig,
 		imageKubeConfig:         imageKubeConfig,
+		gatheringRulesEndpoint:  gatheringRulesEndpoint,
+		httpClient: &http.Client{
+			Transport: httpcache.NewMemoryCacheTransport(),
+			Timeout:   time.Minute,
+		},
```

Comment:
Just note. It would be probably good to move it to `insightsclient` once there's some auth required.